node/src/blockchain/blockchain_interface/blockchain_interface_web3/utils.rs

```diff
@@ -36,72 +42,46 @@ pub struct BlockchainAgentFutureResult {
     pub transaction_fee_balance: U256,
     pub masq_token_balance: U256,
 }
-pub fn advance_used_nonce(current_nonce: U256) -> U256 {
-    current_nonce
-        .checked_add(U256::one())
-        .expect("unexpected limits")
-}
 
-fn error_with_hashes(
-    error: Web3Error,
-    hashes_and_paid_amounts: Vec<HashAndAmount>,
-) -> PayableTransactionError {
-    let hashes = hashes_and_paid_amounts
-        .into_iter()
-        .map(|hash_and_amount| hash_and_amount.hash)
-        .collect();
-    PayableTransactionError::Sending {
-        msg: error.to_string(),
-        hashes,
-    }
+fn return_sending_error(sent_txs: &Vec<Tx>, error: &Web3Error) -> LocalPayableError {
+    LocalPayableError::Sending(
+        sent_txs
+            .iter()
+            .map(|sent_tx| FailedTx::from_sent_tx_and_web3_err(sent_tx, error))
+            .collect(),
+    )
 }
 
-pub fn merged_output_data(
+pub fn return_batch_results(
+    txs: Vec<Tx>,
     responses: Vec<web3::transports::Result<Value>>,
-    hashes_and_paid_amounts: Vec<HashAndAmount>,
-    accounts: Vec<PayableAccount>,
-) -> Vec<ProcessedPayableFallible> {
-    let iterator_with_all_data = responses
-        .into_iter()
-        .zip(hashes_and_paid_amounts.into_iter())
-        .zip(accounts.iter());
-    iterator_with_all_data
-        .map(
-            |((rpc_result, hash_and_amount), account)| match rpc_result {
-                Ok(_rpc_result) => {
-                    // TODO: GH-547: This rpc_result should be validated
-                    ProcessedPayableFallible::Correct(PendingPayable {
-                        recipient_wallet: account.wallet.clone(),
-                        hash: hash_and_amount.hash,
-                    })
-                }
-                Err(rpc_error) => ProcessedPayableFallible::Failed(RpcPayableFailure {
-                    rpc_error,
-                    recipient_wallet: account.wallet.clone(),
-                    hash: hash_and_amount.hash,
-                }),
-            },
-        )
-        .collect()
+) -> BatchResults {
+    txs.into_iter().zip(responses).fold(
+        BatchResults::default(),
+        |mut batch_results, (sent_tx, response)| {
+            match response {
+                Ok(_) => batch_results.sent_txs.push(sent_tx), // TODO: Validate the JSON output
+                Err(rpc_error) => batch_results
+                    .failed_txs
+                    .push(FailedTx::from_sent_tx_and_web3_err(&sent_tx, &rpc_error)),
```

Comment:
Now I'm thinking, in my PR, these elementary conversions exist there, I employed the From trait, naturally; I hope we will end up save those instead of these after the merges happen.